airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py

```diff
@@ -861,7 +873,10 @@ def state(self) -> Mapping[str, Any]:
 
     @state.setter
     def state(self, value: Mapping[str, Any]):
-        self._cursor_value = value[self.cursor_field]
+        if value:
```

Comment:
> Actually, testing manually makes me believe that upping the CDK will cause issues with ClientSideIncrementalStream as the state will be set and this will fail with an index error. I think it comes from a change here where the condition before was if stream_state and "state" in dir(stream_instance) and <...> but we've removed the first part

@maxi297 good comment about this in the `poetry.lock` file about CDK compatibility! So I had adjusted this condition because there's actually kind of a nasty bug in the CDK that could cause data loss for the first slice because the availability strategy can accidentally update state. See this comment in a previous [PR](https://github.com/airbytehq/airbyte/pull/37429/files#r1578493847).

And because of that change, I adjusted this condition. I think Hubspot configures setters a bit uncharacteristic from others and with RFR, we can't assume state setter/getters == incremental and will always have `cursor_field`. Basically to account for the case you mentioned, I updated our state setter to be more resilient to empty state which is what we should probably be doing regardless.